tracing-appender/src/rolling.rs

```diff
@@ -558,17 +569,73 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            keep_last,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, keep_last: usize) {
+        let files = fs::read_dir(&self.log_directory).map(|dir| {
+            dir.filter_map(Result::ok)
+                .filter(|entry| {
+                    entry.file_name().to_string_lossy().contains(
+                        &self
+                            .log_filename_prefix
+                            .clone()
+                            .expect("prefix is always present"),
+                    )
+                })
+                .collect::<Vec<_>>()
+        });
+        let mut files = match files {
+            Ok(files) => files,
+            Err(error) => {
+                eprintln!("Error reading the log directory/files: {}", error);
+                return;
+            }
+        };
+        if files.len() < keep_last {
+            return;
+        }
+        // sort the files by their creation timestamps.
+        files.sort_by_key(|file| {
+            file.metadata()
+                .and_then(|metadata| metadata.created().map(Some))
```

Comment:
I think you could use `map` instead of `and_then` directly

Reply:
I was thinking of this:
```rust
// sort the files by their creation timestamps.
        files.sort_by_key(|file| {
            match file.metadata().and_then(|metadata| metadata.created()) {
                Ok(val) => val,
                Err(why) => {
                    println!("failed reading the timestamp because: {}", why);
                    // if we couldn't determine the file's creation timestamp
                    // for whatever reason, we should return early
                    return;
                }
            };
        });
```

Reply:
1 `and_then()` is beneficial imo, since we have to check both `metadata()` and then `created()`. 
Hahaha I even used the words **and then** myself, what a great language <3

Reply:
Oh, indeed `file.metadata().and_then(|metadata| metadata.created())` looks good! I overlooked the final output was actually an `Option<SystemTime>` 👍

Reply:
Seems like this wouldn't work, because the `return` is not making the function return, but making the match block return